src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -196,30 +207,32 @@ public MemoryPoolIterator ConsumingStart()
                 else
                 {
                     returnStart = _head;
```

Comment:
Never mind. I forgot we use _consuming in dispose. Good job continuing to checking this in the lock.